bigbluebutton-html5/imports/ui/components/audio/container.jsx

```diff
@@ -73,19 +74,20 @@ class AudioContainer extends React.Component {
 
 let didMountAutoJoin = false;
 
-export default withModalMounter(injectIntl(withTracker(({ mountModal, intl }) => {
+export default lockContextContainer(withModalMounter(injectIntl(withTracker(({ mountModal, intl, userLocks }) => {
   const APP_CONFIG = Meteor.settings.public.app;
   const KURENTO_CONFIG = Meteor.settings.public.kurento;
-
   const autoJoin = getFromUserSettings('autoJoin', APP_CONFIG.autoJoin);
+  const { userWebcam } = userLocks;
   const openAudioModal = () => new Promise((resolve) => {
     mountModal(<AudioModalContainer resolve={resolve} />);
   });
 
   const openVideoPreviewModal = () => new Promise((resolve) => {
+    if (userWebcam) return resolve();
     mountModal(<VideoPreviewContainer resolve={resolve} />);
   });
-  if (Service.audioLocked()
+  if (userWebcam
```

Comment:
Why did we switch from checking for mics to checking for webcam in AudioContainer? Is this about the order of auto-join processes?

Reply:
My bad